baselines/flwr_baselines/publications/adaptive_federated_optimization/main.py

```diff
@@ -0,0 +1,75 @@
+"""Runs AdaptiveFederated Optimization for CIFAR10/100."""
+from os import chdir
+from pathlib import Path
+
+import flwr as fl
+import hydra
+from flwr.common.typing import Parameters
+from hydra.utils import call, get_original_cwd, instantiate, to_absolute_path
+from omegaconf import DictConfig
+
+
+@hydra.main(config_path="conf/cifar10", config_name="config")
+def main(cfg: DictConfig) -> None:
+    """General-purpose main function that receives cfg from Hydra."""
+
+    # Make sure we are on the right directory.
+    # This will not be necessary in hydra 1.3
+    chdir(get_original_cwd())
+
+    # Create federated partitions - checkout the config files for details
+    path_original_dataset = Path(to_absolute_path(cfg.root_dir))
+    fed_dir = call(
+        cfg.gen_federated_partitions, path_original_dataset=path_original_dataset
+    )
+
+    # Get centralized evaluation function - see config files for details
+    eval_fn = call(cfg.get_eval_fn, path_original_dataset=path_original_dataset)
+
+    # Define client resources and ray configs
+    client_resources = {"num_cpus": cfg.cpus_per_client}
```

Comment:
So training doesn't use GPU?

Reply:
Solved by including `"num_gpus": cfg.gpus_per_client` and adding `num_gpus: 1` on `config.yaml` files

Reply:
This will assign one entire GPU to a single client (queuing the rest of the clients in the round -- assuming there's only one GPU in the system). Maybe it's worth specifying in the config the amount of VRAM a client needs and then workout what's the GPU ratio that that amount of memory represents? (similar to other projects we've used GPU+CPU resources w/ Ray). Alternatively, I'd say setting `num_gpus: 0.33` should be fine for most systems ...